services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -104,7 +126,92 @@ def _replace_env_vars_in_compose_spec(
     return stringified_service_spec
 
 
-def assemble_spec(
+async def _inject_resource_limits_and_reservations(
+    app: FastAPI, service_key: str, service_tag: str, service_spec: ComposeSpecLabel
+) -> None:
+    director_v0_client: DirectorV0Client = get_director_v0_client(app)
```

Comment:
TIP: This function is only used in ``assemble_spec`` below and has two responsibilities
- GET labels from ``directorv0`` service 
- updates ``service_spec``

Therefore 
- I would apply separation of concerns principle and split it in two.
- I would use standard prefix as ``update`` or ``get`` or ``set`` instead of ``assemble``, ``inject`` ...  I think the former are well defined and helps clarity.

Reply:
This function no longer requires to get labels, ad was changed.